File: src/KeePassLib2Android/Interfaces/ITimeLogger.cs
/*
  KeePass Password Safe - The Open-Source Password Manager
  Copyright (C) 2003-2017 Dominik Reichl <dominik.reichl@t-online.de>

  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program; if not, write to the Free Software
  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
*/

using System;

namespace KeePassLib.Interfaces
{
	/// <summary>
	/// Interface for objects that support various times (creation time, last
	/// access time, last modification time and expiry time). Offers
	/// several helper functions (for example a function to touch the current
	/// object).
	/// </summary>
	public interface ITimeLogger
	{
		/// <summary>
		/// The date/time when the object was created.
		/// </summary>
		DateTime CreationTime
		{
			get;
			set;
		}

		/// <summary>
		/// The date/time when the object was last modified.
		/// </summary>
		DateTime LastModificationTime
		{
			get;
			set;
		}

		/// <summary>
		/// The date/time when the object was last accessed.
		/// </summary>
		DateTime LastAccessTime
		{
			get;
			set;
		}

		/// <summary>
		/// The date/time when the object expires.
		/// </summary>
		DateTime ExpiryTime
		{
			get;
			set;
		}

		/// <summary>
		/// Flag that determines if the object does expire.
		/// </summary>
		bool Expires
		{
			get;
			set;
		}

		/// <summary>
		/// Get or set the usage count of the object. To increase the usage
		/// count by one, use the <c>Touch</c> function.
		/// </summary>
		ulong UsageCount
		{
			get;
			set;
		}

		/// <summary>
		/// The date/time when the location of the object was last changed.
		/// </summary>
		DateTime LocationChanged
		{
			get;
			set;
		}

		/// <summary>
		/// Touch the object. This function updates the internal last access
		/// time. If the <paramref name="bModified" /> parameter is <c>true</c>,
		/// the last modification time gets updated, too. Each time you call
		/// <c>Touch</c>, the usage count of the object is increased by one.
		/// </summary>
		/// <param name="bModified">Update last modification time.</param>
		void Touch(bool bModified);
	}
}

File: src/java/KP2AKdbLibrary/app/src/main/java/com/keepassdroid/database/ITimeLogger.java
/*
 * Copyright 2010 Brian Pellin.
 *     
 * This file is part of KeePassDroid.
 *
 *  KeePassDroid is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  KeePassDroid is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with KeePassDroid.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package com.keepassdroid.database;

import java.util.Date;

public interface ITimeLogger {
	Date getLastModificationTime();
	void setLastModificationTime(Date date);
	
	Date getCreationTime();
	void setCreationTime(Date date);
	
	Date getLastAccessTime();
	void setLastAccessTime(Date date);
	
	Date getExpiryTime();
	void setExpiryTime(Date date);
	
	boolean expires();
	void setExpires(boolean exp);
	
	long getUsageCount();
	void setUsageCount(long count);
	
	Date getLocationChanged();
	void setLocationChanged(Date date);

}
